usersexercise/src/main/java/com/Controllers/UsersController.java

```diff
@@ -0,0 +1,72 @@
+package com.Controllers;
+
+import com.Model.Users;
+import com.Services.UserService;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.beans.BeanUtils;
+import com.Model.Response.UserResponse;
+import com.dto.UserDto;
+import com.Model.Request.UserRequest;
+import java.util.ArrayList;
+import java.util.List;
+
+@RestController
+@RequestMapping("users")
+public class UsersController {
+
+    private final UserService userservice;
+
+    public UsersController(UserService userservice){
+        this.userservice = userservice;
+    }
+
+    @GetMapping(path = "/allUsers")
+    public List<UserResponse> getAllUsers(){
+        List<UserResponse> returnValue = userservice.getAllUsers();
+        return returnValue;
+    }
+
+    @GetMapping
+    public List<UserDto> getUsers(@RequestParam(value = "page", defaultValue = "1") int page,
+                                  @RequestParam(value = "limit", defaultValue = "3") int limit){
+        List<UserDto> returnValue = userservice.getUsers(page, limit);
+        return returnValue;
+    }
+
+    @PostMapping
+    public UserResponse createUser(@RequestBody UserRequest userRequest){
+        UserDto userdto = new UserDto();
```

Comment:
You could do your email validation here without having to do all of the logic and calling the service layer